ReactNativeClient/lib/joplin-renderer/MdToHtml/rules/video.js

```diff
@@ -0,0 +1,29 @@
+// LOGIC: Extract the domain name from the URL(attr: href of link_open)
+// & UUID of the youtube video and render iframe is domain is youtube
+// and the given youtube link has a valid UUID
+
+module.exports = function installRule(markdownIt) {
+
+	const defaultRender = markdownIt.renderer.rules.link_open;
+	const videoIDRegex = /^.*(youtu\.be\/|v\/|u\/\w\/|embed\/|watch\?v=|&v=)([^#&?]*).*/;
+	const key = /.*(youtube|youtu.be).*/im;
+	// videoIDRegex extracts the unique 11 alpha-numeric code which is unique to every youtube video
+
+	markdownIt.renderer.rules.link_open = (tokens, idx, options, env, self) => {
+		const token = tokens[idx];
+		const link = token.attrGet('href');
+		const match = key.test(link);
+		const videoID = link.match(videoIDRegex);
+
+		if (!match || !videoID) return defaultRender(tokens, idx, options, env, self);
+
+		if (match && videoID[2].length == 11) {
```

Comment:
`videoID[2].length == 11` is not necessary if your regex is correct.

Reply:
it's just to prevent faulty youtube links, but I'll remove it if you say so

Reply:
> it's just to prevent faulty youtube links, but I'll remove it if you say so

It can't be faulty if the regex is correct. You could enforce a number of characters in your regex, although I don't think you should in case YouTube change the ID length.

Reply:
>It can't be faulty if the regex is correct. You could enforce a number of characters in your regex, although I don't think you should in case YouTube change the ID length.

okay I'll remove it

***

By faulty links, I meant if someone accidentally changes the video ID of the link(a typo)

![regex](https://user-images.githubusercontent.com/54576074/77446427-68826a00-6e14-11ea-88ab-c2dfaa6f8d00.gif)

It does seem a bit redundant, it was used in other youtube video-embeddings plugins that's why I added it. 


#### REGEX 

<img width="660" alt="regex" src="https://user-images.githubusercontent.com/54576074/77447158-5a811900-6e15-11ea-8bfa-67799c45274a.png">

Reply:
where can I find this REGEX editor?

Reply:
https://www.debuggex.com/